File: src/helpers/faucet/faucet-requester.ts
import axios from 'axios';
import { Config } from '../../../config/default';
import { logger } from '../../utils/logger';
import { Urls } from '../common/urls';
import { FaucetRequest } from './faucet-request';

export const requestFaucet = async (wallet: string, amount: number) => {
  const data: FaucetRequest = { wallet: wallet, amount: amount };
  logger.notice(`URL: ${Config.baseUrl + Urls.request_erc20_tokens}`);
  logger.notice(data);
  const result = await axios.post(
    Config.baseUrl + Urls.request_erc20_tokens,
    data
  );
  logger.notice(result);
};

File: src/helpers/faucet/faucet_requester.py
import requests
from config.config import BASE_URL
from src.helpers.common.urls import Urls
from src.helpers.faucet.faucet_request import FaucetRequest


async def request_faucet(wallet: str, amount: int):
    data: FaucetRequest = FaucetRequest(amount=amount, wallet=wallet)
    print(data)
    url: str = BASE_URL + Urls.REQUEST_ERC20_TOKENS
    requests.post(url, data)
